### ISSUE TITLE
Notice: Undefined index: view in components/com_content/router.php on line 73

### ISSUE THREAD

**Issue by StefanSTS** [joomla/joomla-cms#10721]
#### Steps to reproduce the issue
Open an article by clicking on a menu link.
Article should have a link in content like:
<a href="index.php?option=com_content&amp;view=article&amp;id=14:tutorial-xyz&amp;catid=8:tutorials-de&amp;lang=de-DE&amp;Itemid=133">Produktvariante</a>

#### Expected result
Show the page without notices.

#### Actual result
Shows the notice.

#### System information (as much as possible)
Joomla 3.5.1
PHP 7.0.6
Apache 2.4.7-1ubuntu4.9

#### Additional comments
I tried to reproduce the notice and it showed up for every link inside of an article. So one link, one message, two links, two equal messages. And so on.
Searching for the error message shows various sites with the same problem. Of course switching off the messages would help, but since it shows in combination with links inside the content, there might be something to start with finding the issue.

Regards
Stefan

**Comment by StefanSTS** [comment]
Link got eaten up by the editor, put some sqare brackets not. It's shortend by cutting of the slug a bit.
[a href="index.php?option=com_content&amp;view=article&amp;id=14:tutorial-virtu&amp;catid=8:tutorials-de&amp;lang=de-DE&amp;Itemid=133"]Produktvariante[/a]
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10721">issues.joomla.org/joomla-cms/10721</a>.</sub>

**Comment by ggppdk** [comment]
Hi

The particular IF statement that produces this notice, assumes that the given menu item **to be used for creating the SEF link**,
**always** has variables:
- 'view'
- 'id'

Also it does not care of the menu item is really com_content,
so besides the notice,
- in rare cases an non - com_content menu item can be activated  (if 'view' and 'id' variables of current menu item are a much but point to a different)

----

A relevant discussion, describing the above is here:
https://github.com/joomla/joomla-cms/pull/6488

I am guessing that you did this:
- you added a link in an editor area (e.g. in article description) linking to another article

and the inserted link had been added with menu item id:    133

then at later time you have changed the type menu-item for menu-item 133 ?

**Comment by StefanSTS** [comment]
Hello Georgios,
indeed this was caused by the change of a menu item from category blog to menu item alias (which was pointing to the first child menu item). The link before had the menu item id 133 and the alias was pointing to the child menu item with id 185 or so. Both links had got view=article and (article_)id 13, only the category id and menu item id changed.

The error message made me think, it was a problem with the view=article and probably redirected my attention from the real problem, the category id and menu item id.

I added the links in the article with JCE which put the non SEF link. With putting a SEF link instead, I could have avoided the notice.

I guess only an additional table, that would record all internal links given in articles and then referencing them with the given menu items would help in this case. All links would have to be updated on change of the menu item automatically. 

Thanks for your help to figure out the problem and sorry for bringing up an issue, that might have been better placed in the forum.
Stefan
